controller/podcontroller.go

```diff
@@ -229,31 +221,31 @@ func (c *PodController) checkReadyStatus(oldpod, newpod *apiv1.Pod, cluster *crv
 
 					operator.UpdatePghaDefaultConfigInitFlag(c.PodClientset, false, clusterName, newpod.ObjectMeta.Namespace)
 
-					// update pgreplica status if performing a restore
+					// if pod is coming ready after a restore, create the initial backup instead
+					// of the stanza
 					if cluster.Status.State == crv1.PgclusterStateRestore {
-						pgreplicaList := &crv1.PgreplicaList{}
-						selector := config.LABEL_PG_CLUSTER + "=" + clusterName
-						log.Debugf("Restored cluster %s went to ready, patching replicas", clusterName)
-						err := kubeapi.GetpgreplicasBySelector(c.PodClient, pgreplicaList, selector, newpod.ObjectMeta.Namespace)
+
+						//look up the backrest-repo pod name
+						selector := fmt.Sprintf("%s=%s,pgo-backrest-repo=true", 
+							config.LABEL_PG_CLUSTER, clusterName)
+						pods, err := kubeapi.GetPods(c.PodClientset, selector, 
+							newpod.ObjectMeta.Namespace)
+						if len(pods.Items) != 1 {
+							log.Errorf("pods len != 1 for cluster %s", clusterName)
+							return
+						}
 						if err != nil {
 							log.Error(err)
-							return
```

Comment:
Similarly is there a reason we're not returning here?

I would suggest a comment around why if that's the case.